lib/generate.js

```diff
@@ -68,6 +68,10 @@ function transformValue(value, field, spec) {
 				value = value.replace(nonAsciiRegex, '*');
 			if (!spec.supportsNewLineChars)
 				value = value.replace(newLineRegex, ' ');
+			if (!spec.supportsExtraRecordEndingChars) {
+				var recordEndingRegex = new RegExp(`[${spec.recordEnding}]`, 'g');
```

Comment:
Using a regex in this way is problematic. The characters in `recordEnding` aren't escaped, so it could behave unexpectedly and also introduces a DoS risk. For the specific behaviour (replacing any instance of any of the characters with an asterisk), use of regexes is not necessary. Recommend changing to work without a regex to avoid these two issues.